src/tracing/js/mod.rs

```diff
@@ -428,7 +428,12 @@ where
             depth: context.journal_ref().depth() as u64,
             refund: interp.control.gas().refunded() as u64,
             error: None,
-            contract: self.active_call().contract.clone(),
+            contract: Contract {
+                caller: interp.input.caller_address,
+                contract: interp.input.target_address,
+                value: self.active_call().contract.value,
+                input: self.active_call().contract.input.clone(),
```

Comment:
we can get rid of one  active_call here, because all this does is .last() but still